lib/git_reflow/git_server/git_hub/pull_request.rb

```diff
@@ -78,6 +78,64 @@ def approved?
           end
         end
 
+        def merge!(options = {})
+
+          # fallback to default merge process if user "forces" merge
+          if(options[:force])
+            super options
+          else
+            options[:destination_branch] ||= 'master'
+            feature_branch_name = GitReflow.current_branch
+
+            commit_title = ask("Please enter your delivery commit title: (leaving blank will use default)")
+            commit_message = ask("Please enter your delivery commit message: (leaving blank will use default)")
+
+            if commit_title.strip == ""
+              commit_title = self.title
+            end
+
+            if commit_message.strip == ""
+              commit_message = self.body
+            end
+
+            message = "#{options[:message]}"
+
+            if "#{self.number}".length > 0
+              message << "\nMerges ##{self.number}\n"
+            end
+
+            if lgtm_authors = Array(self.approvals) and lgtm_authors.any?
+              message << "\nLGTM given by: @#{lgtm_authors.join(', @')}\n"
+            end
+
+            data = {
+              "commit_title" => "#{commit_title}",
+              "commit_message" => "#{commit_message}",
+              "sha" => "#{self.head.sha}",
+              "squash" => true
+            }
+            
+            merge_response = GitReflow::GitServer::GitHub.connection.pull_requests.merge(
+              "#{GitReflow.git_server.class.remote_user}", 
+              "#{GitReflow.git_server.class.remote_repo_name}", 
+              "#{self.number}", 
+              data
+            )
+
+            if merge_response.success?
+              GitReflow.run_command_with_label "git checkout #{options[:destination_branch]}"
```

Comment:
let's also `GitReflow.run_command_with_label "git pull #{options[:destination_branch]}"` to grab the merge that was made.